Add Gardenlet deployment docs (#117)
gardener/documentation: website/documentation/guides/install_gardener/setup-seed/_index.md

---

> title:  Register a Seed Cluster

Reviewer: ```suggestion
title: Register a Seed Cluster
```

aliases: ["/readmore/seed", "/050-tutorials/content/howto/setup-seed"]
---

# Deprecated

> Please note that in the meanwhile there is an easier way to register a Shoot cluster as a Seed. 

Reviewer: ```suggestion
Note that there is already an easier way to register a Shoot cluster as a Seed. 
```

These clusters are called `Shooted Seeds` in the Gardener terminology. Please [see this document](https://github.com/gardener/gardener/blob/master/docs/usage/shooted_seed.md) for more information.

To register Seed cluster in "fenced" environments (e.g. behind a firewall), please refer to [this document](https://gardener.cloud/documentation/guides/install_gardener/deploy-gardenlet/).

> # Seed Cluster created by the landscape setup

Reviewer: This heading seems reads as title on github. On the other hand the title in front-matter is "Register a Seed Cluster" and this is confusing.
If I get it right this section is motivation for laying out the procedure that follows. 
Either use the Title as h1 heading (which will make it redundant with the current website documentation pages layout and doesn't look nice, but that's something we have to figure in general), or use something like Motivation or Introduction for header or just skip the header at all (either of which may not look nice if you read it on github because the title will be missing)

> The [landscape-setup-template](https://github.com/gardener/landscape-setup-template) is meant to provide an as-simple-as-possible Gardener installation. 

Reviewer 2: This should refer to garden-setup instead of the landscape-setup-template. The template is horribly outdated.

Therefore it just registers the cluster where the Gardener is deployed on as a seed cluster. 
While this is easy, depending on the underlying cluster, it might be insecure. 
Clusters created with Kubify don't have network policies, for example. 

> See [Hardening the Gardener Community Setup](/readmore/hardening) for more information. 

Reviewer: ```suggestion
See [Hardening the Gardener Community Setup](/documentation/guides/install_gardener/secure-setup/) for more information. 
```

> To have network policies on the seed cluster and avoid having the seed on the same cluster as the Gardener, the easiest option is probably to simply create a shoot and then register that shoot as seed. 

Reviewer: ```suggestion
To have network policies on the seed cluster and avoid having the seed on the same cluster as the Gardener, the easiest option is to simply create a shoot and then register that shoot as seed. 
```

This way you can also leverage other advantages of shooted clusters for your seed, e.g. autoscaling.

## Setting up the Shoot

The first step is to create a shoot cluster.
You can find templates for the shoot manifest [here](https://github.com/gardener/gardener/tree/master/example). 
 

> Note that the Seed cluster currently needs to have different network CIDRs for pods, services and nodes compared to its Shoot clusters.

Reviewer: ```suggestion
Note that the Seed cluster needs to have different network CIDRs for pods, services and nodes compared to its Shoot clusters.
```

      ...
```

## Registering the Shoot as Seed

The seed itself is a Kubernetes resource that can be deployed via a yaml file, but it has some dependencies. 
You can find templated versions of these files in the [seed-config component](https://github.com/gardener/landscape-setup/tree/0.5.0/components/seed-config) of the landscape-setup-template project. 
If you have set up your Gardener using this project, there should also be rendered versions of these files in the `state/seed-config/` directory of your landscape folder (they are probably easier to work with). 
Examples for all these files can also be found in the aforementioned example folder in the Gardener repo. 

### 1. Seed Namespace 

First, you should create a namespace for your new seed and everything that belongs to it. 

> This is not necessary, but it will keep your cluster organized. 

Reviewer: ```suggestion
This is not a requirement, but it will keep your cluster organized. 
```

For this example, the namespace will be called `seed-test`.

### 2. Cloud Provider Secret

> The Gardener needs to create resources on the seed and thus needs a kubeconfig for it. 

Reviewer: ```suggestion
The Gardener needs to create resources on the seed and requires a kubeconfig to be able to do that. 
```
